db/migrate/20230303165757_create_upload_snapshots.rb

```diff
@@ -1,7 +1,9 @@
 class CreateUploadSnapshots < ActiveRecord::Migration[6.1]
   def change
     create_table :upload_snapshots do |t|
-      t.string :uri, null: false
+      t.string :filename
+      t.string :url
+      t.bigint :version
```

Comment:
Carolyn's imagination....
For each work you would have many entires:
{ files:[
   {filename:  'name1', checksum: 'abc1232'},
   {filename:  'name2', checksum: 'abc1232'},
   {filename:  'name3', checksum: 'abc1232'},
]
}